### PR TITLE
use chat protocol and contacts in chat commands/messages

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: apps/simplex-chat/Simplex/Chat/protocol.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Chat protocol
+
+## Message syntax
+
+The syntax of the message inside agent MSG:
+
+```abnf
+agentMessageBody = message / msgContinuation
+message = [chatMsgId] SP msgEvent SP [parameters] SP msgBody SP
```
Comment (efim-poberezkin) [inline apps/simplex-chat/Simplex/Chat/protocol.md:9]: why have `SP` in the end of `message`? wouldn't surrounding delimiters on an envelope would be enough for parsing?

File: apps/simplex-chat/Simplex/Chat/protocol.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+txtMsgData = <utf-8 encoded text>
+jsonMsgData = <JSON as utf-8 encoded text>
+partSize = 1*DIGIT ; size of this part within this message
+totalPartSize = 1*DIGIT ; total size of this part across all messages, including the current one
+msgEventParents = msgEventParent *msgEventParent ; binary body part for content type "prev"
+msgEventParent = memberId refMsgId refMsgHash
+memberId = 8*8(OCTET) ; shared member ID
+refMsgId = 8*8(OCTET) ; sequential message number - external agent message ID
+refMsgHash = 16*16(OCTET) ; SHA256 of agent message body
+msgContinuation = "#" prevMsgId SP partSize SP msgData *(SP msgBodyPart)
```
Comment (efim-poberezkin) [inline apps/simplex-chat/Simplex/Chat/protocol.md:40]: what is `msgContinuation` for? is it for large messages?
should the first `partSize` and `msgData` also be `msgBodyPart`?

File: apps/simplex-chat/Simplex/Chat/protocol.md (hunk 1 above)
Comment (epoberezkin) [reply apps/simplex-chat/Simplex/Chat/protocol.md]: at the point of parsing the envelop is gone - it's just `MSG MsgMeta MsgBody`. So again, it's easier to parse when this delimiter is allowed (but it is currently optional anyway)

File: apps/simplex-chat/Main.hs [hunk 3]
```diff
@@ -40,11 +41,11 @@ logCfg = LogConfig {lc_file = Nothing, lc_stderr = True}
 main :: IO ()
 main = do
   ChatOpts {dbFile, smpServers} <- welcomeGetOpts
-  void $ createStore "simplex-chat.db" 4
+  st <- createStore (dbFile <> ".chat.db") 4
   ct <- newChatTerminal
-  a <- getSMPAgentClient cfg {dbFile, smpServers}
+  a <- getSMPAgentClient cfg {dbFile = dbFile <> ".agent.db", smpServers}
```
Comment (efim-poberezkin) [inline apps/simplex-chat/Main.hs:46]: with default db path wouldn't these become `...smp-chat.db.chat.db` and `...smp-chat.db.agent.db`? change default to `combine appDir "simplex"`?

File: apps/simplex-chat/Simplex/Chat.hs [hunk 4]
```diff
@@ -61,66 +72,136 @@ inputSubscriber = do
...
+    conns <- withStore $ \st -> getContactConnections st userId cRef
+    withAgent cRef $ \smp -> forM_ conns $ \Connection {agentConnId} ->
+      deleteConnection smp agentConnId `catchError` \(_ :: AgentErrorType) -> pure ()
+    void $ withStore $ \st -> deleteContact st userId cRef
+    unsetActive $ ActiveC cRef
+    when (null conns) . throwError . ChatErrorContact $ CENotFound cRef
+    showContactDeleted cRef
+  SendMessage cRef msg -> do
+    userId <- asks currentUserId
+    Connection {agentConnId} <- withStore $ \st -> getContactConnection st userId cRef
```
Comment (efim-poberezkin) [inline apps/simplex-chat/Simplex/Chat.hs:102]: in DeleteContact all contact connections are deleted - how is it defined which connection to use in SendMessage?

File: apps/simplex-chat/Simplex/Chat.hs [hunk 5]
```diff
@@ -61,66 +72,136 @@ inputSubscriber = do
...
+    let body = MsgBodyContent {contentType = SimplexContentType XCText, contentHash = Nothing, contentData = MBFull $ MsgData msg}
+        rawMsg = rawChatMessage ChatMessage {chatMsgId = Nothing, chatMsgEvent = XMsgNew MTText, chatMsgBody = [body], chatDAGIdx = Nothing}
+    void . withAgent cRef $ \smp -> sendMessage smp agentConnId $ serializeRawChatMessage rawMsg
+    setActive $ ActiveC cRef
```
Comment (efim-poberezkin) [inline apps/simplex-chat/Simplex/Chat.hs:106]: is it correct that notion of active contact is needed only for terminal chat (and not other UI)? if so shouldn't it be decoupled from chat logic, e.g. by a way of terminal component changing it?

**Review by efim-poberezkin — APPROVED** [review]

File: apps/simplex-chat/Main.hs (hunk 3 above)
Comment (epoberezkin) [reply apps/simplex-chat/Main.hs:46]: Yes, I didn’t check the default

File: apps/simplex-chat/Simplex/Chat.hs (hunk 4 above)
Comment (epoberezkin) [reply apps/simplex-chat/Simplex/Chat.hs:102]: The one that’s marked as active. Currently there is no implemented scenario when contact can have more Han one connection, the first created one is marked as activr

File: apps/simplex-chat/Simplex/Chat.hs (hunk 5 above)
Comment (epoberezkin) [reply apps/simplex-chat/Simplex/Chat.hs:106]: I would worry about it later

**Pull request merged**